### PR TITLE
Added privateEndpoint bicep template(s)

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Change

- Added privateEndpoint bicep template(s)
- Added RBAC
- Added lock
- Removed ARM template
- Pipeline reference
  [Network: Privateendpoints](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.privateendpoints.yml)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -92,7 +92,7 @@ This repository includes a collection of advanced and curated Modules consisting
...
+| [PrivateEndpoints](https://github.com/Azure/ResourceModules/tree/main/arm/Microsoft.Network/privateEndpoints) | :heavy_check_mark: | [![Network: Privateendpoints](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.privateendpoints.yml/badge.svg)](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.privateendpoints.yml) | [![Deploy to Azure](/docs/media/deploytoazure.svg?sanitize=true)](<https://portal.azure.com/#create/Microsoft.Template/uri/https%3a%2f%2fraw.githubusercontent.com%2fAzure%2fResourceModules%2fmain%2farm%5cMicrosoft.Network%5cprivateEndpoints%2fdeploy.json>) |
```
Comment (eriqua) [inline README.md:95]: I guess we should remove the deploy to azure button if deploy.json is deleted

File: arm/Microsoft.Network/privateEndpoints/deploy.bicep [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  params: {
+    roleAssignmentObj: roleAssignment
+    builtInRoleNames: builtInRoleNames
+    resourceName: privateEndpoint.name
+  }
+}]
+
+output privateEndpointResourceGroup string = resourceGroup().name
+output privateEndpointResourceId string = privateEndpoint.id
+output privateEndpointName string = privateEndpointName
```
Comment (eriqua) [inline arm/Microsoft.Network/privateEndpoints/deploy.bicep:109]: Same as the other 2: should we go with `privateEndpointName` or `privateEndpoint.name`?

File: testResults.xml [hunk 3]
```diff
@@ -0,0 +1,223 @@
+﻿<?xml version="1.0" encoding="utf-8" standalone="no"?>
```
Comment (eriqua) [inline testResults.xml:1]: Are we adding this file?

**Comment by eriqua** [comment]
Also in this case the readme content cannot be loaded 
image

File: README.md (hunk 1 above)
Comment (MrMCake) [reply README.md:95]: True, and I'll update it. However, I'd not focus too much on the table as we have the auto-generating pipeline ready in the fork

Comment (MrMCake) [reply README.md]: done

**Pull request merged**